docs/source/advanced/jit.rst

```diff
@@ -206,6 +206,29 @@ A simple constant optimization example::
     The XLA backend (the interface to the hardware) will do the constant optimization and may take a long time and
     a lot of memory due to compilation, often with very little gain in final performance, if any.
 
+RandomState and Jit
+^^^^^^^^^^^^^^^^^^^
+
+You need to pass :code:`objax.random.DEFAULT_GENERATOR` as a variable to :code:`VarCollection`,
+otherwise the :code:`DEFAULT_GENERATOR` will be treated as a constant and the jitted function
+will always return the same value.
+
+.. code-block:: python
+
+    import objax
+    objax.random.DEFAULT_GENERATOR.seed(123)
+
+
+    def function():
```

Comment:
Recommended style is use `Function.with_vars` instead of passing variable collection to JIT:

```python
@objax.Function.with_vars(objax.random.DEFAULT_GENERATOR.vars())
def function():
    d = objax.random.uniform((1,))
    return d

function_jit = objax.Jit(function)
```

Reply:
updated